### PR TITLE
CLOUDP-244167: atlasdatabaseuser: use indexer instead of resourcewatcher

### PR DESCRIPTION
**Pull request by s-urbaniak**
### All Submissions:

* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes-template.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

**Review by josvazg — APPROVED** [review]
LGTM, ok to merge when CI is also green

File: pkg/indexer/atlasdatabaseusersecrets.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import (
+	"go.uber.org/zap"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+)
+
+const (
+	AtlasDatabaseUserBySecretsIndex = "atlasdatabaseuser.spec.passwordSecret"
```
Comment (helderjs) [inline pkg/indexer/atlasdatabaseusersecrets.go:11]: dbUser belongs to a project, should we add an index for it too?
Deployments would benefit it when creating connection secret, it goes through all DBUsers to match the ones that belong to the same project in order to create the respective connection secret

**Review by helderjs — COMMENTED** [review]
Overall LGTM, just raised a question once index for DBUser is being introduced.

**Review by helderjs — APPROVED** [review]

File: pkg/indexer/atlasdatabaseusersecrets.go (hunk 1 above)
Comment (s-urbaniak) [reply pkg/indexer/atlasdatabaseusersecrets.go:11]: yes, that makes sense. the database user controller reads the project so we should add an index on it and let the atlas database user reconciler watch projects as well.

This is not part of this PR though, this mechanically replaces resource watcher with indexers.

**Pull request merged**